api/services/reporting/report_generator.py

```
@@ -0,0 +1,211 @@
+            html_content = template_obj.render(**context)
+
+            # Generate PDF in thread pool (blocking operation)
+            loop = asyncio.get_event_loop()
+            pdf_bytes = await loop.run_in_executor(
+                None,
+                self._generate_pdf_sync,
+                html_content
+            )
+
+            logger.info("PDF generated successfully (%d bytes)", len(pdf_bytes))
+            return pdf_bytes
```

Ruff (0.14.5)

Consider moving this statement to an `else` block

(TRY300)